### PR TITLE
Add rubocop

### PR DESCRIPTION
**Pull request by eduardoj**
- Add rubocop configuration file.
- Fix all offenses with auto-correct.
- Fix manually Lint/UselessAssignment and Metrics/LineLength rubocop offenses.
- Add files to .gitignore to make rake package work.
- Add rubygem-docker-api package to Dockerfile.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# use the shared Yast defaults
+inherit_from:
+  - /usr/share/YaST2/data/devtools/data/rubocop_yast_style.yml
+  - .rubocop_todo.yml
```
Comment (jreidinger) [inline .rubocop.yml]: I think it is better to merge stuff from todo here, so it is not spread over two files. At least this is what other modules do.

**Comment by jreidinger** [comment]
beside one note this looks good, but please also fix conflict.

File: .rubocop.yml (hunk 1 above)
Comment (Ana06) [reply .rubocop.yml]: The `.rubocop.yml` is your configuration, so what you want to change from the default configuration, and the `.rubocop_todo.yml` is what you want to enable, but didn't have enable yet. This todo file is used for avoiding having to solve many offenses when adding Rubocop to a big project. So this is perfectly fine.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,29 @@
+.yardoc/checksums
```
Comment (Ana06) [inline .gitignore]: Why do you need to add all this to the gitignore? 🤔

**Comment by eduardoj** [comment]
@jreidinger:
> beside one note this looks good, but please also fix conflict.

Conflict fixed.

Regarding using `.rubocop_todo.yml` file I agree with @Ana06: I prefer using this separate file instead of including the offenses in the `.rubocop.yml` file. Can this also be considered as a generic rule for all the yast modules from now on?

**Comment by jreidinger** [comment]
ok, so approved. Good to see rubocop there.

**Comment by jreidinger** [comment]
Do you have permissions to merge?

**Comment by eduardoj** [comment]
`Do you have permissions to merge?`

No.

**Comment by jreidinger** [comment]
@eduardoj and do you maintain now module instead of @flavio or additionally to him or is it just one time project?

**Comment by eduardoj** [comment]
@jreidinger Just one time project.

**Comment by jreidinger** [comment]
OK, so I merged it myself.

**Pull request merged**